insights/parsers/audit_log.py

```diff
@@ -0,0 +1,133 @@
+"""
+audit_log - File ``/var/log/audit/audit.log``
+=============================================
+
+"""
+
+import shlex
+from datetime import date
+from .. import LogFileOutput, parser
+
+filter_list = [
+    'AVC',
+    'denied',
+    'comm="mongod"'
+]
+
+
+@parser("audit_log", filter_list)
+class AuditLog(LogFileOutput):
+    """
+    Class for parsing ``/var/log/audit/audit.log`` file.
+
+    Sample log lines::
+
+        type=CRYPTO_KEY_USER msg=audit(1506046832.641:53584): pid=16865 uid=0 auid=0 ses=7247 subj=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023 msg='op=destroy kind=session fp=? direction=both spid=16865 suid=0 rport=59296 laddr=10.72.32.205 lport=22  exe="/usr/sbin/sshd" hostname=? addr=10.66.136.139 terminal=? res=success'
+        type=LOGIN msg=audit(1506047401.407:53591): pid=482 uid=0 subj=system_u:system_r:crond_t:s0-s0:c0.c1023 old-auid=4294967295 auid=993 old-ses=4294967295 ses=7389 res=1
+        type=AVC msg=audit(1506487181.009:32794): avc:  denied  { create } for  pid=27960 comm="mongod" scontext=system_u:system_r:mongod_t:s0 tcontext=system_u:system_r:mongod_t:s0 tclass=unix_dgram_socket
+
+    Examples:
+        >>> log = shared[AuditLog]
+        >>> log.get('type=AVC')
+        'type=AVC msg=audit(1506487181.009:32794): avc:  denied  { create } for  pid=27960 comm="mongod" scontext=system_u:system_r:mongod_t:s0 tcontext=system_u:system_r:mongod_t:s0 tclass=unix_dgram_socket'
+        >>> avc_log_info = list(log.parse_lines(log.get('type=AVC')))[0]
+        >>> avc_log_info
+        {
+            'timestamp': '1506487181.009',
+            'unparsed': 'avc:  denied  { create } for',
+            'msg_ID': '32794',
+            'pid': '27960',
+            'raw_message': 'type=AVC msg=audit(1506487181.009:32794): avc:  denied  { create } for  pid=27960 comm="mongod" scontext=system_u:system_r:mongod_t:s0 tcontext=system_u:system_r:mongod_t:s0 tclass=unix_dgram_socket',
+            'comm': 'mongod',
+            'scontext': 'system_u:system_r:mongod_t:s0',
+            'tclass': 'unix_dgram_socket',
+            'type': 'AVC',
+            'tcontext': 'system_u:system_r:mongod_t:s0'
+        }
+        >>> assert avc_log_info == list(log.get_after(timestamp=date.fromtimestamp(1506047401.407)))[0]
+    """
+
+    def parse_lines(self, lines):
```

Comment:
As we discussed, this `parse_lines` should not be a public interface of this class, otherwise it could accept any kinds of `lines` and parse them as the log of `audit.log`

Reply:
Changed.